src/cart/crt_iv.c

```diff
@@ -1066,10 +1071,16 @@ handle_ivfetch_response(const struct crt_cb_info *cb_info)
 
 	IV_DBG(&input->ifi_key, "response received, rc = %d\n", rc);
 
+	/* Inline IV fetch */
+	if (iv_info->ifc_bulk_hdl == NULL)
+		d_sgl_buf_copy(&iv_info->ifc_iv_value, &output->ifo_sgl);
```

Comment:
i dont think we should copy if rc != 0. maybe you can rework check on line 1061-1064, this and subsequent check in on_refresh()   (rc == 0 ? : ) and instead just pass null iv_value on error and a valid one otherwise (along with an inline copy if needed)

Reply:
sure